### PR TITLE
Fix having to pass in -f flag to docker-compose

### PR DESCRIPTION
**Pull request by murny**
Also lots of refactors to conform to bash best practices

This fixes Number 1 in this issue https://github.com/ualbertalib/jupiter/issues/168

> When we pull down docker-compose.deployment.yml onto the server, we should rename this to just docker-compose.yml then you no longer have to pass the flag option with the file (-f docker-compose.deployment.yml) to the docker-compose command everywhere

### CODE REVIEW

File: deploy [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```
Comment (murny) [inline deploy]: Executables shouldn't have an extension, also provide a shebang and space this script out a bit for readability

File: deploy [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+docker rmi $(docker images -f "dangling=true" -q)
+
+docker-compose pull web
```
Comment (murny) [inline deploy]: No longer need the `-f docker-compose.deployment.yml` everywhere as we are pulling down this yml file and renaming it to the default name

File: predeploy [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# This script should be either rsynced or copied to the UAT server when setting up fresh server.
+# For more information, please see the documentation: https://github.com/ualbertalib/di_internal/blob/master/System-Adminstration/UAT-Environment.md
+
+set -o errexit
+set -o pipefail
+set -o nounset
+[[ "${TRACE}" ]] && set -o xtrace
+
+
+readonly branch_name="master"
```
Comment (murny) [inline predeploy]: Best practice is to use upper case for Environment variables and also to make these variables immutable

File: predeploy [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+readonly install_directory="/home/deploy/jupiter"
+
+mkdir -p "${install_directory}"
+mkdir -p "${install_directory}/config"
```
Comment (murny) [inline predeploy]: Use quotes and `{}` everywhere as bash can do tricky things when you have spaces, etc.

File: predeploy [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+curl -o "${install_directory}/deploy" "https://raw.githubusercontent.com/ualbertalib/jupiter/${branch_name}/deploy"
+curl -o "${install_directory}/docker-compose.yml" "https://raw.githubusercontent.com/ualbertalib/jupiter/${branch_name}/docker-compose.deployment.yml"
+curl -o "${install_directory}/config/nginx.conf" "https://raw.githubusercontent.com/ualbertalib/jupiter/${branch_name}/config/nginx.conf"
+
+if [[ ! -f "${install_directory}/.env_deployment" ]]
```
Comment (murny) [inline predeploy]: Previously this if check wasn't doing what it was intended. As it was checking the directory of this script and not the actual install_directory (which very well could be different!)

File: predeploy [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+[[ "${TRACE}" ]] && set -o xtrace
+
+
+readonly branch_name="master"
+readonly install_directory="/home/deploy/jupiter"
+
+mkdir -p "${install_directory}"
+mkdir -p "${install_directory}/config"
+curl -o "${install_directory}/deploy" "https://raw.githubusercontent.com/ualbertalib/jupiter/${branch_name}/deploy"
+curl -o "${install_directory}/docker-compose.yml" "https://raw.githubusercontent.com/ualbertalib/jupiter/${branch_name}/docker-compose.deployment.yml"
```
Comment (murny) [inline predeploy]: Main change of this PR. Let's rename the `docker-compose.deployment.yml` to `docker-compose.yml` when we pull it down from github

File: predeploy [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# This script will be sitting in /home/deploy directory, and is for setting up directories and pulling required files for docker deployment
+# This script should be either rsynced or copied to the UAT server when setting up fresh server.
+# For more information, please see the documentation: https://github.com/ualbertalib/di_internal/blob/master/System-Adminstration/UAT-Environment.md
+
+set -o errexit
```
Comment (murny) [inline predeploy]: Good practice to set these

**Comment by weiweishi** [comment]
LGTM. Learnt something new.
I totally missed the ticket #168 but agreed that many of the points should be addressed to make the script/process more clear to read and more streamlined to execute.

**Pull request merged**